### DISCUSSION TITLE
Some question about the Program plugin

### DISCUSSION THREAD

**Discussion by VictoriousRaptor** [Flow-Launcher/Flow.Launcher#1434] [Q&A] [answered]
Some designs of Program plugin don't make sense for me. I can't understand why it's designed this way, and hope someone could tell me.

1. What is the purpose of the "All Programs" button? If I add a directory and click it, it shows every single program under the directory as an item.
2. The design of the "Delete/Disable/Enable" 3-in-one button. I've check the code, the function `ProgramSetting.Row_OnClick()` is not called when I click a row. So the button text is always "Disable". Not sure why. But why use this 3-in-one design? It's confusing.
3. If I accidentally click the "All Programs" button, I don't how to delete the items created. The only way I found to delete them is to manually edit settings json.
4. Performance of this plugin. I remember the Program plugin of Wox took ages to init. I haven't encounter this issue in Flow version. Did you optimize it?

**Comment by jjw24** [comment, marked as answer]
1. Showing all programs helps managing them because it displays all programs loaded by the plugin, from the list which you can then disable/enable certain programs.

**Comment by jjw24** [comment]
2. Are you using the pre-release version? If so I think there is a bug which is not changing the state of the button. Prod release 1.9.5 shouldn't have this problem I believe.
3 in 1 is due to back then space for each plugin was limited and to save space, and thought it was cool to have button change base on the state of functionality. Happy to take feedback and improve the design.

**Reply by VictoriousRaptor** [reply]
1.9.5 has this issue too.

**Reply by jjw24** [reply]
Ok, will track this issue in https://github.com/Flow-Launcher/Flow.Launcher/issues/1356

**Comment by jjw24** [comment]
3. If you clicked all programs button, you should have a button that shows 'Enable' if you select on one of the entry.

**Reply by VictoriousRaptor** [reply]
I hope to delete all the entries created. Too many of them.

**Reply by jjw24** [reply]
You won't be able to delete, you can disable them from being displayed, which will achieve the intended decluttering goal.

**Comment by jjw24** [comment]
4. Yes we have significantly optimized this plugin and as well as in general the way plugins are managed and executed.
